src/util/system.cpp

```diff
@@ -121,9 +124,10 @@ std::vector<std::string> get_mapped_paths_for_pid(pid_t pid)
   std::vector<std::string> paths;
 
   // start with the exe
-  std::string pid_exe = get_pid_exe(pid);
-  if (!pid_exe.empty() && pid_exe.find("(deleted)") == std::string::npos)
-    paths.push_back(get_pid_exe(pid));
+  auto pid_exe = get_pid_exe(pid);
+  if (pid_exe && !pid_exe->empty() &&
```

Comment:
IIUC, we can drop the `!pid_exe->empty()` condition as we either return a path or an error. Occurs in multiple places.